sdk/storage/azdatalake/file/client_test.go

```diff
@@ -1337,6 +1339,169 @@ func (s *UnrecordedTestSuite) TestFileDeleteWithSAS() {
+
+	srcFileClient, err := file.NewClientWithNoCredential(fsClient.DFSURL()+"/file?"+sasToken, nil)
+	_require.NoError(err)
+	_require.NotNil(srcFileClient)
+
+	resp, err := srcFileClient.Create(context.Background(), nil)
+	_require.NoError(err)
+	_require.NotNil(resp)
+
+	// create local file
+	_, content := generateData(10 * 1024)
+	err = os.WriteFile("testFile", content, 0644)
+	handleError(err)
+
+	defer func() {
+		err = os.Remove("testFile")
+		handleError(err)
+	}()
+
+	fh, err := os.Open("testFile")
+	handleError(err)
+
+	defer func(fh *os.File) {
+		err := fh.Close()
+		handleError(err)
+	}(fh)
+
+	// upload the file
+	err = srcFileClient.UploadFile(context.Background(), fh, &file.UploadFileOptions{
+		Concurrency: 5,
+		ChunkSize:   2 * 1024,
+	})
+	handleError(err)
+	defer testcommon.DeleteFileSystem(context.Background(), _require, fsClient)
+
+	response, err := srcFileClient.DownloadStream(context.Background(), nil)
+	handleError(err)
+	testcommon.DeleteFileSystem(context.Background(), _require, fsClient)
+
+	_require.Equal(encryptionScope, *response.EncryptionScope)
+}
+
+func (s *UnrecordedTestSuite) TestGetUserDelegationEncryptionScopeSAS() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDatalake, nil)
+	_require.NoError(err)
+
+	filesystemName := testcommon.GenerateFileSystemName(testName)
+	fsClient := testcommon.CreateNewFileSystem(context.Background(), _require, filesystemName, svcClient)
+
+	defer testcommon.DeleteFileSystem(context.Background(), _require, fsClient)
+
+	_, err = testcommon.GetRequiredEnv(testcommon.DataLakeEncryptionScopeEnvVar)
+	_require.Nil(err)
+
+	// Set current and past time and create key
+	currentTime := time.Now().UTC().Add(-10 * time.Second)
+	pastTime := currentTime.Add(48 * time.Hour)
+	info := service.KeyInfo{
+		Start:  to.Ptr(currentTime.UTC().Format(sas.TimeFormat)),
+		Expiry: to.Ptr(pastTime.UTC().Format(sas.TimeFormat)),
+	}
+
+	udc, err := svcClient.GetUserDelegationCredential(context.Background(), info, nil)
+	_require.NoError(err)
+
+	// get permissions and details for sas
+	encryptionScope, err := testcommon.GetRequiredEnv(testcommon.DataLakeEncryptionScopeEnvVar)
+	_require.Nil(err)
+
+	// Create Blob Signature Values with desired permissions and sign with user delegation credential
+	perms := sas.FilePermissions{Read: true, Create: true, Write: true, Move: true, Delete: true, List: true}
+	sasQueryParams, err := sas.DatalakeSignatureValues{
+		Protocol:        sas.ProtocolHTTPS, // Users MUST use HTTPS (not HTTP)
+		StartTime:       time.Now().UTC().Add(time.Second * -10),
+		ExpiryTime:      time.Now().UTC().Add(15 * time.Minute), // 15 minutes before expiration
+		FileSystemName:  filesystemName,
+		Permissions:     perms.String(),
+		EncryptionScope: encryptionScope,
+	}.SignWithUserDelegation(udc)
+	_require.Nil(err)
+
+	sasURL := fmt.Sprintf(fsClient.DFSURL() + sasQueryParams.Encode())
```

Comment:
Shouldn't this be, `sasURL := fsClient.DFSURL() + "/file?" + sasQueryParams.Encode()`?